frontend/src/components/UserContext.tsx

```diff
@@ -0,0 +1,15 @@
+import React, { createContext } from 'react'
+import { fetchUser } from '../hooks/fetchUser'
+
+export const UserContext = createContext<any>(null)
+
+export const GetUserContext = ({ children }: any): JSX.Element => {
+  const user = fetchUser()
```

Comment:
shouldn't this be an async function requiring await? i haven't done this specific thing you're doing here so I might be wrong

Reply:
Yeah i didn't find the need to use an await here cause the custom hook is not returning a promise, plus the useEffect callback fn in `fetchUser()` is taking care of async API request

Reply:
hmm okay i won't block on this issue but something here feels off - at the very least can you add some more explicit typings here instead of a bunch of `any`?

Reply:
added explicit typings